### PR TITLE
Always use external CoinUtils, Osi and Clp libraries

### PR DESCRIPTION
**Pull request by p12tic**
Previously we used an internal fork of CoinUtils, Osi and Clp libraries that was included as a submodule. The repository structure was significantly different than what upstream has (3 repositories vs. 1 on our side) and thus tracking upstream was harder than necessary.

This PR converts the project to use new forks of upstream CoinUtils, Osi and Clp at https://github.com/alicevision/CoinUtils, https://github.com/alicevision/Osi and https://github.com/alicevision/Clp. These repositories currently have pending PRs with cmake support.

### CODE REVIEW

**Comment by simogasp** [comment]
[note to self] we will also need to update the docker images as they are used for the ci

**Comment by servantftechnicolor** [comment]
Very cool, this task was in my backlog. Will reduce the rebuild time of alicevision.

**Comment by fabiencastan** [comment]
I'm rebuilding the docker image for the dependencies to be able to use it in the linux CI.

**Comment by p12tic** [comment]
I've rebased on top of latest `develop` branch and pinned the dependencies to the exact commits like is done for the rest of dependencies.

**Comment by fabiencastan** [comment]
I get this error when building it with `./docker/build-centos.sh`:
```
CMake Error at /usr/local/share/cmake-3.21/Modules/BundleUtilities.cmake:471 (file):
  file READ_ELF given FILE "libClp.so" that does not exist.
Call Stack (most recent call first):
  /usr/local/share/cmake-3.21/Modules/BundleUtilities.cmake:527 (get_item_rpaths)
  /usr/local/share/cmake-3.21/Modules/BundleUtilities.cmake:649 (set_bundle_key_values)
  /usr/local/share/cmake-3.21/Modules/BundleUtilities.cmake:934 (get_bundle_keys)
  /opt/AliceVision_git/src/cmake/MakeBundle.cmake:138 (fixup_bundle)
```

**Comment by p12tic** [comment]
@fabiencastan I've fixed the `fixup_bundle` error. The install prefix was not specified.

**Comment by fabiencastan** [comment]
We will need to add osi to vcpkg.

**Comment by simogasp** [comment]
> We will need to add osi to vcpkg.

https://github.com/microsoft/vcpkg/tree/master/ports/osi

File: .github/workflows/continuous-integration.yml [hunk 1]
```diff
@@ -174,6 +174,8 @@ jobs:
                           ceres[suitesparse,cxsparse]
                           tbb
                           assimp
+                          clp
```
Comment (simogasp) [inline .github/workflows/continuous-integration.yml:177]: clp will automatically import `coinutils` and `osi`

**Comment by fabiencastan** [comment]
```
-- FLANN: 1.8.4 (internal)
-- CLP: 
-- COINUTILS: 
-- OSI: 
-- LEMON:  (internal)
```

Missing XX_VERSION variables.

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -990,23 +925,9 @@ if(ALICEVISION_BUILD_SFM)
...
-  endif()
-
-  if(ALICEVISION_USE_INTERNAL_OSI)
-    message(STATUS "OSI: " ${OSI_VERSION} " (internal)")
-  else()
-    message(STATUS "OSI: " ${OSI_VERSION} " (external)")
-  endif()
+  message(STATUS "CLP: " ${CLP_VERSION} "")
+  message(STATUS "COINUTILS: " ${COINUTILS_VERSION} "")
+  message(STATUS "OSI: " ${OSI_VERSION} "")
```
Comment (simogasp) [inline src/CMakeLists.txt:930]: probably?
```suggestion
  message(STATUS "CLP: " ${Clp_VERSION} "")
  message(STATUS "COINUTILS: " ${CoinUtils_VERSION} "")
  message(STATUS "OSI: " ${osi_VERSION} "")
```

Comment (simogasp) [reply src/CMakeLists.txt:930]: https://cmake.org/cmake/help/latest/command/find_package.html#config-mode-version-selection

Comment (simogasp) [reply src/CMakeLists.txt]: I confirm that it must be the same case as the name in the `find_package`, so `${Osi_VERSION}, ${Clp_VERSION}, ${CoinUtils_VERSION}`

Comment (fabiencastan) [reply src/CMakeLists.txt]: It works fine, thanks.

**Review by fabiencastan — APPROVED** [review]

**Pull request merged**